File: solutions/00001/python/main.py
if __name__ == "__main__":
    multiples_of_3 = {x for x in range(3, 1000, 3)}
    multiples_of_5 = {x for x in range(5, 1000, 5)}
    all_multiples = multiples_of_3.union(multiples_of_5)
    total_sum = sum(all_multiples)
    print(total_sum)

File: solutions/00001/typescript/main.ts
function *range(a: number, b?: number, step: number = 1): Generator<number> {
  for (let i = a; !b || i < b; i += step) {
    yield i
  }
}

if (import.meta.main) {
  const multiplesOf3 = new Set(Array.from(range(3, 1000, 3)))
  const multiplesOf5 = new Set(Array.from(range(5, 1000, 5)))
  const allMultiples = multiplesOf3.union(multiplesOf5)
  const multipleSum = Array.from(allMultiples).reduce((a, b) => a + b)
  console.log(multipleSum)
}
